utils/bimap/bimap.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bimap
+
+import (
+	"sync"
+
+	"github.com/ava-labs/avalanchego/utils"
+)
+
+var _ BiMap[int, int] = (*biMap[int, int])(nil)
+
+type Entry[K, V any] struct {
+	Key   K
+	Value V
+}
+
+// BiMap is a bi-directional map.
+type BiMap[K, V any] interface {
```

Comment:
Should we just export the struct and get rid of the interface?

Reply:
I'm ok with a struct... If we add something like a metered map or something then we can re-add the interface.